adapters/repos/db/index.go

```diff
@@ -887,6 +898,12 @@ func (i *Index) IncomingBatchPutObjects(ctx context.Context, shardName string,
 		}
 	}
 
+	shard, release, err := i.getOrInitLocalShardNoShutdown(ctx, shardName)
+	if err != nil {
+		return duplicateErr(ErrShardNotFound, len(objects))
```

Comment:
same comment as before, probably it's better to return `err` to avoid hiding a different error that could be returned by `getOrInitLocalShardNoShutdown`

Reply:
resolve to merge